### PR TITLE
kill unwanted parent directories when a segment is deleted from LocalDataSegmentKiller

### LINKED ISSUE #2272: Segments parent directory not removed on permanent deletion with NFS deep-storage
I have setup a Druid cluster on five KVM (CentOs 6.5-based) on our private Joyent cloud.
The cluster executse the Coordinator, Historical, Overlord, Broker and Realtime node each one on a separate VM, whilst the middleManager is on the same VM of the Overlord (even if configured with druid.indexer.runner.type=remote).
Deep storage is configured on NFS (provided on NetApp appliance).
Using the "old coordinator console", I am able to disable a datasource and then to permanently delete its segments; however parent directories of removed segments are not removed from the deep storage.
As an example, if I have this segment-related file:
/NFS/sso024/2015-08-13T00\:00\:00.000Z_2015-08-14T00\:00\:00.000Z/2016-01-13T14\:30\:02.644Z/0/index.zip
after the deletion I have:
/NFS/sso024/2015-08-13T00\:00\:00.000Z_2015-08-14T00\:00\:00.000Z/2016-01-13T14\:30\:02.644Z/
Notice that killing tasks are marked "SUCCESSFULL" in the Overlord Web UI.

### PR DESCRIPTION
**Pull request by himanshug**
fixes #2272

### CODE REVIEW

File: server/src/main/java/io/druid/segment/loading/LocalDataSegmentKiller.java [hunk 1]
```diff
@@ -36,49 +37,33 @@
   public void kill(DataSegment segment) throws SegmentLoadingException
   {
     final File path = getDirectory(segment);
-    log.info("segment[%s] maps to path[%s]", segment.getIdentifier(), path);
+    try {
```
Comment (fjy) [inline server/src/main/java/io/druid/segment/loading/LocalDataSegmentKiller.java]: can we add a log line that we are killing a segment located at location <somethign>

**Comment by fjy** [comment]
:+1: even if my comment isn't addressed

File: server/src/main/java/io/druid/segment/loading/LocalDataSegmentKiller.java (hunk 1 above)
Comment (himanshug) [inline server/src/main/java/io/druid/segment/loading/LocalDataSegmentKiller.java]: added

File: server/src/main/java/io/druid/segment/loading/LocalDataSegmentKiller.java [hunk 2]
```diff
@@ -35,50 +36,36 @@
...
-      return;
-    }
+    try {
+      if (path.getName().endsWith(".zip")) {
 
-    final File[] files = path.listFiles();
-    int success = 0;
+        // path format -- > .../dataSource/interval/version/partitionNum/xxx.zip
+        File partitionNumDir = path.getParentFile();
+        FileUtils.deleteDirectory(partitionNumDir);
```
Comment (navis) [inline server/src/main/java/io/druid/segment/loading/LocalDataSegmentKiller.java]: Should we delete parent directory only when it's empty? Or at least there should be an option for it.

Comment (himanshug) [inline server/src/main/java/io/druid/segment/loading/LocalDataSegmentKiller.java]: partition directory corresponds to the specific segment being deleted. it should always be deleted.

**Comment by nishantmonu51** [comment]
:+1:, looks good to me.

File: server/src/main/java/io/druid/segment/loading/LocalDataSegmentKiller.java [hunk 3]
```diff
@@ -35,50 +36,36 @@
...
   {
-    final Map<String, Object> loadSpec = segment.getLoadSpec();
-    final File path = new File(MapUtils.getString(loadSpec, "path"));
-
-    if (!path.exists()) {
-      throw new SegmentLoadingException("Asked to load path[%s], but it doesn't exist.", path);
-    }
-
-    return path.getParentFile();
+    return new File(MapUtils.getString(segment.getLoadSpec(), "path"));
```
Comment (drcrallen) [inline server/src/main/java/io/druid/segment/loading/LocalDataSegmentKiller.java]: Is there any way to tie this to LoadLocalSpec ? parsing the loadspec and calling getPath seems a little extreme for here, but can "path" be put as a `public static final String` in the class or something?

Comment (himanshug) [inline server/src/main/java/io/druid/segment/loading/LocalDataSegmentKiller.java]: made it static var

**Comment by drcrallen** [comment]
:+1: after travis

**Comment by himanshug** [comment]
@fjy @nishantmonu51 @drcrallen  can one of you merge this one?

**Pull request merged**